siliconcompiler/schema/schema_cfg.py

```diff
@@ -3507,96 +3595,185 @@ def schema_asic(cfg):
+            example=[
+                "cli: -constraint_component_placement 'i0 (2.0,3.0,0.0)'",
+                "api: chip.set('constraint', 'component', 'i0', 'placement', (2.0,3.0,0.0)"],
+            schelp="""
+            Placement location of a named component, specified as a (x,y,z) tuple of
+            floats. The location refers to the placement of the center/centroid of the
+            component. The 'placement' parameter is a goal/intent, not an exact specification.
+            The compiler and layout system may adjust sizes to meet competing
+            goals such as manufacturing design  rules and grid placement
+            guidelines. The 'z' coordinate shall be set to 0 for planar systems
+            with only (x,y) coordinates. Discretized systems like PCB stacks,
+            package stacks, and breadboards only allow a reduced
+            set of floating point values (0,1,2,3). The user specifying the
+            placement will need to have some understanding of the type of
+            layout system the component is being placed in (ASIC, SIP, PCB) but
+            should not need to know exact manufacturing specifications.""")
+
+    scparam(cfg, ['constraint', 'component',  instance, 'rotation'],
+            sctype='float',
+            shorthelp="Constraint: Component rotation",
+            switch="-constraint_component_rotation 'inst <float>'",
+            example=[
+                "cli: -constraint_component_rotation 'i0 90'",
+                "api: chip.set('constraint', 'component', 'i0', 'rotation', '90')"],
+            schelp="""
+            Placement rotation of the component specified in degrees. Rotation
+            goes counter-clockwise for all parts on top and clock-wise for parts
+            on the bottom. In both cases, this is from the perspective of looking
+            at the top of the board. Rotation is specified in degrees. Most gridded
+            layout systems (like ASICs) only allow a finite number of rotation
+            values (0,90,180,270).""")
+
+    scparam(cfg, ['constraint', 'component', instance, 'flip'],
+            sctype='bool',
+            shorthelp="Constraint: Component flip option",
+            switch="-constraint_component_flip 'inst <bool>'",
+            example=[
+                "cli: -constraint_component_flip 'i0 true'",
+                "api: chip.set('constraint', 'component', 'i0', 'flip', 'true')"],
+            schelp="""
+            Boolean parameter specifying that the instanced library component should be flipped
+            around the vertical axis befong being placed on the substrate. The need to
+            flip a component depends on the component footprint. Most dies have pads
+            facing up and so must be flipped when assembled face down (eg. flip-chip,
+            WCSP).""")
+
+    # PINS
+    scparam(cfg, ['constraint', 'pin', instance, 'placement'],
+            sctype='(float,float,float)',
```

Comment:
How would this work on an ASIC design? there the natural thing would be to set the `z` parameters to the metal name. Or would you not be able to constrain that here and instead use the `sc_hpinmetal/sc_vpinmetal`?

Reply:
Added to schema